src/vos/vos_obj.c

```diff
@@ -41,10 +41,95 @@ D_CASSERT((uint32_t)VOS_VIS_FLAG_VISIBLE == (uint32_t)EVT_VISIBLE);
 D_CASSERT((uint32_t)VOS_VIS_FLAG_PARTIAL == (uint32_t)EVT_PARTIAL);
 D_CASSERT((uint32_t)VOS_VIS_FLAG_LAST == (uint32_t)EVT_LAST);
 
+static int
+empty_tree_check(daos_handle_t ih, vos_iter_entry_t *entry,
+		 vos_iter_type_t type, vos_iter_param_t *param, void *cb_arg,
+		 unsigned int *acts)
+{
+	bool	*empty = cb_arg;
+
+	*empty = false;
+
+	return 1; /* Return positive number to break iteration */
+}
+
+static int
+tree_is_empty(struct vos_object *obj, daos_handle_t toh,
+	      const daos_epoch_range_t *epr, vos_iter_type_t type)
+{
+	bool	empty = true;
+	int	rc;
+
+	rc = vos_iterate_key(obj, toh, type, epr, empty_tree_check, &empty,
+			     NULL);
+
+
+	if (rc < 0)
+		return rc;
+
+	if (empty)
+		return 1;
+
+	return 0;
+}
+
+static int
+vos_propagate_check(struct vos_object *obj, daos_handle_t toh,
+		    struct vos_ts_set *ts_set, const daos_epoch_range_t *epr,
+		    vos_iter_type_t type)
+{
+	const char	*tree_name = "AKEY";
```

Comment:
Not a big deal, but why set it to "AKEY" initially? Could put in case for VOS_ITER_AKEY to be consistent.

Reply:
fixed